File: golangLeetcode/66.加一.go
/*
 * @lc app=leetcode.cn id=66 lang=golang
 *
 * [66] 加一
 */

// @lc code=start
func plusOne(digits []int) []int {
	scale := 1
	l := len(digits)
	for i := 0; i < l; i++ {
		tmp := digits[l-i-1] + scale
		scale = tmp / 10
		digits[l-i-1] = tmp % 10
		if scale == 0 {
			break
		}
	}
	if scale == 1 {
		res := make([]int, 1)
		res[0] = 1
		digits = append(res, digits...)
	}
	return digits
}

// @lc code=end

File: LoveSwiftLeetcode/66.加一.swift
/*
 * @lc app=leetcode.cn id=66 lang=swift
 *
 * [66] 加一
 */

// @lc code=start
class Solution {
    func plusOne(_ digits: [Int]) -> [Int] {
        var digits = digits
        var left = 1
        var i = 0
        let cnt = digits.count
        while i < cnt {
            let value = digits[cnt - 1 - i] + left
            digits[cnt - 1 - i] = value % 10
            left = value / 10 
            i += 1
        }
        if left == 1 {
            digits.insert(1, at: 0)
        }

        return digits

    }
}
// @lc code=end

